lib/tasks/dev.rake

```diff
@@ -120,63 +107,59 @@ task({ sample_data: :environment }) do
         program_id: program.id,
         creator_id: program.creator_id,
         contact_id: program.creator_id,
-        start_date:,
-        end_date:,
+        start_date: start_date,
+        end_date: end_date,
         required_meetings: [6, 10].sample
       )
       count += 1
     end
   end
 
-  # Creating Program Admins
+# Creating Program Admins
   admins.each do |admin|
     program = Program.find_by(creator_id: admin.id)
+
     ProgramAdmin.create(
       user_id: admin.id,
       program_id: program.id
     )
   end
 
-  # Creating Cohort Members
-  mentors_and_mentees.each do |user|
-    cohort = Cohort.all.sample
-
-    if cohort.nil?
-      puts "No cohort available to assign user: #{user.email}"
-      next
-    end
+# Creating Cohort Members
+mentors_and_mentees.each do |user|
+  cohort = Cohort.all.sample
 
-    role = role_hash[user]
-    capacity = ''
-    if role_hash[user] == 'mentor'
-      role = 'mentor'
-      capacity = rand(2..4)
-    elsif role_hash[user] == 'mentee'
-      role = 'mentee'
-    end
+  if cohort.nil?
+    puts "No cohort available to assign user: #{user.email}"
+    next
+  end
 
-    CohortMember.create(
-      email: user.email,
-      cohort_id: cohort.id,
-      role:,
-      capacity:
-    )
+  role = role_hash[user]
+  capacity = ''
+  if role_hash[user] == 'mentor'
+    role = 'mentor'
+    capacity = rand(2..4)
+  elsif role_hash[user] == 'mentee'
+    role = 'mentee'
   end
 
-  # Creating Matches
+  CohortMember.create(
+    email: user.email,
+    cohort_id: cohort.id,
+    role:,
+    capacity:
+  )
+end
+
+# Creating Matches
   mentees.each do |mentee|
-    cohort_member_mentee = mentee.cohort_members.first
-    if cohort_member_mentee.nil?
-      puts "No cohort member found for mentee with email: #{mentee.email}"
-      next # Skip to the next mentee
-    end
-    mentor_cohort_member_object = CohortMember.where(cohort_id: cohort_member_mentee.cohort_id, role: 'mentor').sample
-    shared_cohort = Cohort.find_by(id: cohort_member_mentee.cohort_id)
-    this_mentor = User.find_by(email: mentor_cohort_member_object.email)
+    cohort_member_id_of_mentee = mentee.cohort_members.first
+    mentor_cohort_member_object = CohortMember.where(cohort_id: cohort_member_id_of_mentee.cohort_id, role: 'mentor').sample
+    shared_cohort = Cohort.find_by(id: cohort_member_id_of_mentee.cohort_id)
     Match.create(
-      mentor_id: this_mentor.id,
+      mentor_id: mentor_cohort_member_object.user_id,
```

Comment:
I could be wrong but won't it be ```mentor_cohort_member_object.user.id``` since the cohort member model doesn't have ```user_id```, but referencing to user model through the email field?

Reply:
Good eye m8! I'll fix before a merge.